src/hooks/useIntersectionObserver.tsx

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useRef, useState } from 'react';
+
+interface UseIntersectionObserverProps {
+  root?: Element | null;
+  rootMargin?: string;
+  threshold?: number | number[];
+}
+
+interface UseIntersectionObserverReturn {
+  ref: React.RefObject<HTMLDivElement>;
+  isIntersecting: boolean;
+}
+
+const useIntersectionObserver = ({
+  root = null,
+  rootMargin = '0px',
+  threshold = 0.1,
+}: UseIntersectionObserverProps): UseIntersectionObserverReturn => {
+  const [isIntersecting, setIsIntersecting] = useState(false);
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      (entries) => {
+        const entry = entries[0];
+        setIsIntersecting(entry.isIntersecting);
+      },
+      {
+        root,
+        rootMargin,
+        threshold,
+      },
+    );
+
+    const currentRef = ref.current;
+    if (currentRef) {
+      observer.observe(currentRef);
+    }
+
+    return () => {
+      if (currentRef) {
+        observer.unobserve(currentRef);
+      }
+    };
+  }, [root, rootMargin, threshold]);
```

Comment:
```suggestion
    const currentRef = ref.current;
    if (!currentRef) {
      return;
    }
    
    const observer = new IntersectionObserver(
      (entries) => {
        const entry = entries[0];
        setIsIntersecting(entry.isIntersecting);
      },
      {
        root,
        rootMargin,
        threshold,
      },
    );
    
    observer.observe(currentRef);

    return () => observer.unobserve(currentRef);
  }, [root, rootMargin, threshold]);
```
그냥 미리 return 해버리면 어떨까요?